modules/cudf/src/series/numeric.ts

```diff
@@ -78,6 +79,25 @@ export abstract class NumericSeries<T extends Numeric> extends Series<T> {
     return Series.new({type: dataType, data: this._col.data, length: newLength});
   }
 
+  /**
+   * Concat a NumericSeries to the end of the caller, returning a new NumericSeries.
+   *
+   * @param other The NumericSeries to concat to the end of the caller.
+   *
+   * @example
+   * ```typescript
+   * import {Series} from '@rapidsai/cudf';
+   *
+   * Series.new([1, 2, 3]).concat(Series.new([4, 5, 6])) // [1, 2, 3, 4, 5, 6]
+   * ```
+   */
+  concat<R extends Numeric>(other: Series<R>,
+                            memoryResource?: MemoryResource): Series<CommonType<T, R>> {
+    const type = CUDF.findCommonType(this.type, other.type);
+    return Series.new(this._col.cast(type, memoryResource)
+                        .concat(other._col.cast(type, memoryResource), memoryResource));
```

Comment:
Small performance optimization suggestion -- we can avoid one or both of the cast ops if the dtypes are already the same as what we're casting to. This leverages the fact that Arrow's `DataType` classes have a `compareTo` method for checking dtype equivalence:

```suggestion
    const type = CUDF.findCommonType(this.type, other.type);
    const lhs = type.compareTo(this.type) ? this._col : this._col.cast(type, memoryResource);
    const rhs = type.compareTo(other.type) ? other._col : other._col.cast(type, memoryResource);
    return Series.new(lhs.concat(rhs, memoryResource));
```